### PR TITLE
DCOS-39076: add sortable table header

### PR DESCRIPTION
**Pull request by mperrotti**
Moved the sortable header cell from staged ui-kit-stage in dcos-ui to ui-kit and added tests

sortableheadercell

### CODE REVIEW

File: packages/table/components/SortableHeaderCell.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+interface IProps {
+  sortHandler: () => void;
+  sortDirection: SortDirection;
+  columnContent: string | React.ReactNode;
+}
+interface IState {
+  hovered: boolean;
+}
+
+export class SortableHeaderCell extends React.Component<IProps, IState> {
```
Comment (mperrotti) [inline packages/table/components/SortableHeaderCell.tsx]: We have a similar component called `HeaderCell`, so I think it makes sense to change the original name, `SortableColumnHeader`. Happy to change back if anybody feels `SortableColumnHeader` is clearer

**Comment by juliangieseke** [comment]
Yay 🎉 - does it work already with right aligned `<HeaderCell />`? 🤔

File: packages/table/components/SortableHeaderCell.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        <HeaderCell
+          onMouseEnter={this.hoverStart}
+          onMouseLeave={this.hoverEnd}
+          className={cx(
+            sortableHeaderIconBaseCSS,
+            styleArrowDirection(displaySortDirection)
+          )}
+          role="columnheader"
+          aria-sort={ariaSortString}
+          aria-label={`sort by ${columnContent} in ${
```
Comment (nLight) [inline packages/table/components/SortableHeaderCell.tsx]: you are my hero!

**Comment by nLight** [comment]
In `dcos-ui` we use `cursor: pointer` for the sortable header cells. Do we want to have it in the `ui-kit` by default?

File: packages/table/tests/__snapshots__/SortableHeaderCell.test.tsx.snap [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+.emotion-0:after {
+  border-color: currentColor transparent transparent transparent;
+  border-bottom-width: 0;
+  border-left-width: 4px;
+  border-right-width: 4px;
+  border-top-width: 5px;
+}
+
+<Clickable
```
Comment (Poltergeist) [inline packages/table/tests/__snapshots__/SortableHeaderCell.test.tsx.snap]: ❤️

**Comment by Poltergeist** [comment]
the `cursor: pointer` CSS we should add to the clickable component instead.

**Comment by juliangieseke** [comment]
I think this particular issue only was about moving the existing element to UI Kit, right aligned functionality is added in https://jira.mesosphere.com/browse/DCOS-41057

**Pull request merged**